src/components/ArticlePage.jsx

```diff
@@ -1,19 +1,18 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { getArticle } from "../../api";
+import { getArticle, patchArticle } from "../../api";
 import "../styles/ArticlePage.css";
 import ArticlesList from "./ArticlesList";
 import CommentsList from "./CommentsList";
 
 function ArticlePage() {
   const [article, setArticle] = useState(null);
-  const [isLoading, setIsLoading] = useState(true);
-
   const { article_id } = useParams();
+  const [hasUpvoted, setHasUpvoted] = useState(false);
+  const [hasDownvoted, setHasDownvoted] = useState(false);
```

Comment:
The way this logic is setup atm, you can either upvote and undo - or - downvote and undo, but not both